_posts/2024-08-18-keyword-from-thesaurus.md

```diff
@@ -0,0 +1,77 @@
+---
+layout: post
+title: Keywords from thesaurus as Queryable
+author: Paul van Genuchten
+author_url: https://github.com/pvangenuchten
+publish_date: 2024-09-18 14:59:00-0400
+---
+
+## Keywords from thesaurus as Queryable
+
+A common convention in catalogues is the use of keywords from a dedicated thesaurus. The assignment of these keywords can then later be used to filter or query the catalogue by these terms. To achieve this use case in pycsw, some configuration needs to be tailored. This blog post indicates the changes needed for this scenario.
+
+For this example we'll use a keyword from the [INSPIRE Themes](http://inspire.ec.europa.eu/theme) thesaurus. We will define a new queryable `inspiretheme`, which will be populated with the relevant keyword (if present).
+
+You can repeat these steps for any other thesaurus.
+
+## Extra database column
+
+Extend the records table in the database with an extra field for the selected thesaurus. This is usually a manual operation on the database.
+
+```sql
+ALTER TABLE records
+ADD inspiretheme VARCHAR(255);
+```
+
+## Add parameter to pycsw
+
+In `pycsw/core/config.py` the newly created database column can be registered to pycsw. 
+
+```python 
+'pycsw:InspireTheme': 'inspiretheme',
+```
+
+## Add column to mapping
+
+`etc/mappings.py` links the pycsw parameter to the columnname in the table.
+
+```python 
+'pycsw:InspireTheme': 'inspiretheme',
+```
+
+## Define parameter as queryable
+
+Which of the parameters are queryable is defined in `pycsw/core/repository.py`.
+
+ 'inspiretheme': self.dataset.inspiretheme,
+
+## Add parameter to record results?
+
+Keywords are already published in records, so there is generally no need to extend the record with the new parameter. If needed you can do so in `pycsw/ogc/api/records.py#Line=1150`.
```

Comment:
If needed you can do so in `pycsw/ogc/api/records.py#Line=1150`. -> If needed you can do so in `pycsw/ogc/api/records.py` (line 1150).